server/container_create_linux.go

```diff
@@ -548,36 +505,13 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 
 		if !isInCRIMounts("/sys", containerConfig.GetMounts()) {
 			specgen.RemoveMount("/sys")
-			specgen.RemoveMount("/sys/fs/cgroup")
-			sysMnt := rspec.Mount{
-				Destination: "/sys",
-				Type:        "sysfs",
-				Source:      "sysfs",
-				Options:     []string{"nosuid", "noexec", "nodev", "ro"},
-			}
-			specgen.AddMount(sysMnt)
+			ctr.SpecAddMount("/sys", "sysfs", "sysfs", "/sys/fs/cgroup", []string{"nosuid", "noexec", "nodev", "ro"})
 		}
 	}
 
 	if ctr.Privileged() {
-		specgen.RemoveMount("/sys")
-		specgen.RemoveMount("/sys/fs/cgroup")
-
-		sysMnt := rspec.Mount{
-			Destination: "/sys",
-			Type:        "sysfs",
-			Source:      "sysfs",
-			Options:     []string{"nosuid", "noexec", "nodev", "rw"},
-		}
-		specgen.AddMount(sysMnt)
-
-		cgroupMnt := rspec.Mount{
-			Destination: "/sys/fs/cgroup",
-			Type:        "cgroup",
-			Source:      "cgroup",
-			Options:     []string{"nosuid", "noexec", "nodev", "rw", "relatime"},
-		}
-		specgen.AddMount(cgroupMnt)
+		ctr.SpecAddMount("sys", "sysfs", "sysfs", "/sys", []string{"nosuid", "noexec", "nodev", "rw"})
```

Comment:
Whoops, seems that the mount destination `/sys` became `sys`. Or is it intentional?

Reply:
fixed!